integration/spark/shared/src/main/java/io/openlineage/spark/agent/util/PathUtils.java

```diff
@@ -75,14 +74,27 @@ public static DatasetIdentifier fromCatalogTable(
 
     Optional<URI> metastoreUri = extractMetastoreUri(sparkConf);
     // TODO: Is the call to "metastoreUri.get()" really needed?
-    //   Java's Optional should prevent the null in the first place.
+    // Java's Optional should prevent the null in the first place.
     if (metastoreUri.isPresent() && metastoreUri.get() != null) {
       // dealing with Hive tables
       DatasetIdentifier symlink = prepareHiveDatasetIdentifier(catalogTable, metastoreUri.get());
       return di.withSymlink(
           symlink.getName(), symlink.getNamespace(), DatasetIdentifier.SymlinkType.TABLE);
+    } else if (catalogTable.provider().isDefined()
+        && sparkConf.isPresent()
+        && "hive".equals(catalogTable.provider().get())
+        && "com.amazonaws.glue.catalog.metastore.AWSGlueDataCatalogHiveClientFactory"
+            .equals(sparkConf.get().get("spark.hadoop.hive.metastore.client.factory.class", ""))) {
+      String region = System.getenv().get("AWS_DEFAULT_REGION");
+      if (region == null || region.isEmpty()) {
+        region = System.getenv("AWS_REGION");
+      }
```

Comment:
Can `region` still can be null in the end or the AWS_REGION is always set?

Reply:
@kacpermuda I've seen that `AWS_DEFAULT_REGION` is supported in all currently supported Glue versions, the `AWS_REGION` seems to be a good fallback given that there is confusion in AWS whether to use one var or another.